lib/vrl/compiler/src/expression/function_closure.rs

```diff
@@ -0,0 +1,182 @@
+use core::ExpressionError;
+
+use crate::expression::Block;
+use crate::parser::{Ident, Node};
+use crate::value::VrlValueConvert;
+use crate::{Context, Expression, Value};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FunctionClosure {
+    variables: Vec<Node<Ident>>,
+    block: Block,
+}
+
+impl FunctionClosure {
+    pub(crate) fn new(variables: Vec<Node<Ident>>, block: Block) -> Self {
+        Self { variables, block }
+    }
+
+    pub fn run_key_value(
```

Comment:
Some function documentation here would be very useful!

Reply:
Resolved in a90a1d0.